### ISSUE TITLE
NullPointerException after opening SATC solution

### ISSUE THREAD

**Issue by IanMayo** [debrief/debrief#1781]
We created an SATC solution.

When I tried to expand it in Outline view, Debrief threw an NPE.

Then, almost whatever we did in Debrief, it threw an NPE.  The Error Log listed the source of the NPE as: AbstractTreeViewer.getSelection()

I guess part of the SATC solution isnt' rendering properly in the Outline view.
## Investigation

I'm unable to reproduce this on my sample data.  On a future visit I'll have to try running several SATC solution (including sample data ones) to try to isolate cause.

**Comment by IanMayo** [comment]
I had this problem again today.  The NPE was apparently on line 2956 of *AbstractTreeViewer.java*.

We need to find a reproducible case.

@theanuradha - could you look out for getting a NullPointerException from AbstractTreeViewer.

I *think* it's arising from the Outline View, but couldn't reproduce it.

**Comment by theanuradha** [comment]
@IanMayo any steps I can try to reproduce issue ? Btw when u get NPW do u have the full stack trace ?

**Comment by IanMayo** [comment]
Hi there, the NPE is an intermittent one.  Next time it occurs in a repeatable way, I'll be sure to catch it.  I had been ignoring it, presuming it wouldn't impact users, but on Tuesday I saw that it did - so it's worth me/us directing some effort towards it.

**Comment by IanMayo** [comment]
Just had the problem, after pressing "Collapse all layers" in the Outline view:
````
!ENTRY org.eclipse.ui 4 0 2017-01-24 11:06:54.502
!MESSAGE Unhandled event loop exception
!STACK 0
java.lang.NullPointerException
	at org.eclipse.jface.viewers.AbstractTreeViewer.getSelection(AbstractTreeViewer.java:2956)
	at org.eclipse.jface.viewers.StructuredViewer.handleSelect(StructuredViewer.java:1211)
	at org.eclipse.jface.viewers.StructuredViewer$4.widgetSelected(StructuredViewer.java:1241)
	at org.eclipse.jface.util.OpenStrategy.fireSelectionEvent(OpenStrategy.java:239)
	at org.eclipse.jface.util.OpenStrategy.access$4(OpenStrategy.java:233)
	at org.eclipse.jface.util.OpenStrategy$1.handleEvent(OpenStrategy.java:403)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1053)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1077)
	at org.eclipse.swt.widgets.Widget.sendSelectionEvent(Widget.java:1094)
	at org.eclipse.swt.widgets.TreeItem.setExpanded(TreeItem.java:1385)
	at org.eclipse.jface.viewers.TreeViewer.setExpanded(TreeViewer.java:332)
	at org.eclipse.jface.viewers.AbstractTreeViewer.internalCollapseToLevel(AbstractTreeViewer.java:1571)
	at org.eclipse.jface.viewers.AbstractTreeViewer.internalCollapseToLevel(AbstractTreeViewer.java:1586)
	at org.eclipse.jface.viewers.AbstractTreeViewer.collapseToLevel(AbstractTreeViewer.java:751)
	at org.eclipse.jface.viewers.AbstractTreeViewer.collapseAll(AbstractTreeViewer.java:733)
	at org.mwc.debrief.core.editors.PlotOutlinePage$12.run(PlotOutlinePage.java:617)
	at org.eclipse.jface.action.Action.runWithEvent(Action.java:498)
	at org.eclipse.jface.action.ActionContributionItem.handleWidgetSelection(ActionContributionItem.java:584)
	at org.eclipse.jface.action.ActionContributionItem.access$2(ActionContributionItem.java:501)
	at org.eclipse.jface.action.ActionContributionItem$6.handleEvent(ActionContributionItem.java:452)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1053)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4169)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3758)
	at org.eclipse.ui.internal.Workbench.runEventLoop(Workbench.java:2701)
	at org.eclipse.ui.internal.Workbench.runUI(Workbench.java:2665)
	at org.eclipse.ui.internal.Workbench.access$4(Workbench.java:2499)
	at org.eclipse.ui.internal.Workbench$7.run(Workbench.java:679)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:668)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:149)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:124)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:110)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:79)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:353)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:180)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:629)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:584)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1438)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1414)
````

**Comment by theanuradha** [comment]
@IanMayo  will analyze

**Comment by IanMayo** [comment]
I was able to inspect it in debug mode.  getSelection was looping through the items in the control.

The first item was a TreeItem (representing a track). Other than that, the array had 50 null objects.  But, SWT expected there to be a real item in there.

**Comment by IanMayo** [comment]
@theanuradha - getting this quite often.  It sometimes needs me to close/restart.

But, I haven't been able to isolate it to a reproducible example.  I think it occurs after adding/removing some items from the Outline view.  I'll keep trying to find a reproducible example.

**Comment by IanMayo** [comment]
Hey, @theanuradha  - at last I have reproducible case, in branch `2086_update_zones`

Steps:
* open **turn.dpf**
* expand **MAYO**
* expand **Sensors**
* expand **Sensor:BowMerchantVessel**
* ctrl-select 3rd to 10th child item
* right-click, and select **Generate TMA solution from selected cuts**
* dialog will open. Click on **Finish**
* click on **Collapse all layers** in Outline pane.

NPE gets thrown:

````

!ENTRY org.eclipse.ui 4 0 2017-02-13 14:29:18.896
!MESSAGE Unhandled event loop exception
!STACK 0
java.lang.NullPointerException
	at org.eclipse.jface.viewers.AbstractTreeViewer.getSelection(AbstractTreeViewer.java:2956)
	at org.eclipse.ui.part.PageBookView$SelectionProvider.getSelection(PageBookView.java:298)
	at org.eclipse.ui.internal.services.WorkbenchSourceProvider.updateSelection(WorkbenchSourceProvider.java:157)
	at org.eclipse.ui.internal.services.WorkbenchSourceProvider.getCurrentState(WorkbenchSourceProvider.java:136)
	at org.eclipse.ui.internal.services.WorkbenchSourceProvider$6.handleEvent(WorkbenchSourceProvider.java:676)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
	at org.eclipse.swt.widgets.Display.filterEvent(Display.java:1262)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1052)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1077)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1058)
	at org.eclipse.swt.widgets.Decorations.WM_ACTIVATE(Decorations.java:1664)
	at org.eclipse.swt.widgets.Shell.WM_ACTIVATE(Shell.java:2145)
	at org.eclipse.swt.widgets.Control.windowProc(Control.java:4525)
	at org.eclipse.swt.widgets.Canvas.windowProc(Canvas.java:341)
	at org.eclipse.swt.widgets.Decorations.windowProc(Decorations.java:1627)
	at org.eclipse.swt.widgets.Shell.windowProc(Shell.java:2069)
	at org.eclipse.swt.widgets.Display.windowProc(Display.java:4976)
	at org.eclipse.swt.internal.win32.OS.PeekMessageW(Native Method)
	at org.eclipse.swt.internal.win32.OS.PeekMessage(OS.java:3129)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3753)
	at org.eclipse.ui.internal.Workbench.runEventLoop(Workbench.java:2701)
	at org.eclipse.ui.internal.Workbench.runUI(Workbench.java:2665)
	at org.eclipse.ui.internal.Workbench.access$4(Workbench.java:2499)
	at org.eclipse.ui.internal.Workbench$7.run(Workbench.java:679)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:668)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:149)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:124)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:110)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:79)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:353)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:180)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:629)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:584)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1438)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1414)
````

Will try to capture video.

**Comment by theanuradha** [comment]
@IanMayo  perfect I will look in to this

**Comment by IanMayo** [comment]
Great - I'm praying it reproduces for you...

**Comment by theanuradha** [comment]
@IanMayo on mac it seems ok will switch to windows

**Comment by IanMayo** [comment]
Ok @theanuradha , here's a video of the above steps:
https://youtu.be/94OLrvWgrvs

**Comment by IanMayo** [comment]
Fixed in #2094
